### DISCUSSION TITLE
METplus 11.x and Intel oneAPI compiler support

### DISCUSSION THREAD

**Discussion by rickgrubin-tomorrow** [dtcenter/METplus#2799] [Installation] [answered]
[This discussion](https://github.com/dtcenter/METplus/discussions/1957) from 2022 notes:

> Thank you for this information. I suspect it will be useful for our users and also useful in updating the compile_MET_all.sh script once Intel creates their new release and deprecates icc and we have access to the new Intel compiler for testing.

The [latest supported Intel oneAPI compiler suites](https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-supported-product-versions.html) (`v2025.0.0`, `v2024.2.1`, `v2023.2.4`) deprecate `icc` in favor of `icx` in versions `v2025.0.0` and `v2024.2.1`.

Further, to the point of opening this topic -- current documentation [MET v11.1.1 and supported compilers](https://met.readthedocs.io/en/latest/Users_Guide/installation.html#required-compilers-and-scripting-languages) states:

> The MET package was developed and tested using the GNU g++/gfortran compilers and the Intel icc/ifort compilers. As additional compilers are successfully tested, they will be added to the list of supported platforms/compilers.

Could you please elaborate on support -- formal or informal or planned -- for `Intel oneAPI C++` (`icpx`) and `Fortran` (`ifx`) compilers?

**Comment by jprestop** [comment, marked as answer]
Hi @rickgrubin-tomorrow. 

> Could you please elaborate on support -- formal or informal or planned -- for Intel oneAPI C++ (icpx) and Fortran (ifx) compilers?

Thank you for your question.  With our upcoming official coordinated release, scheduled for just a few weeks away in December, MET-12.0.0 will include support for the Intel oneAPI compilers.  We have installed our developmental MET-12.0.0 on several machines using the oneAPI compilers.  However, we have been made aware that not all versions of oneAPI are supported at this time. 

We currently know that MET-12.0.0 and its dependent libraries work well with oneAPI versions 2023.2.1, 2022.2.1, and 2022.0.2.  We will be working toward ensuring that our dependent libraries and MET itself will compile well with those more recent versions of the oneAPI compilers, however that support is not planned for version MET-12.0.0.

MET-11.1.0 does not and will not include support for the Intel oneAPI compilers.
